pkg/agent/controller/serviceexternalip/controller.go

```diff
@@ -71,7 +71,7 @@ type ServiceExternalIPController struct {
 
 	queue workqueue.RateLimitingInterface
 
-	externalIPStates      map[apimachinerytypes.NamespacedName]externalIPState
+	externalIPStates      map[apimachinerytypes.NamespacedName]*externalIPState
```

Comment:
is there a reason for making this is a pointer? Based on usage in `GetServiceExternalIPStatus`, it doesn't seem necessary.

Reply:
It is just for the defer statement to save the `externalIPState`. I have reverted this change since it is not necessary and can also be achieved by changing the signature of `func saveServiceState`.